crates/parcel_nodejs/src/parcel_v2/transformer.rs

```diff
@@ -0,0 +1,40 @@
+use napi::Env;
+use napi::JsObject;
+use napi::JsUnknown;
+use napi_derive::napi;
+use parcel_macros::napi::create_macro_callback;
+
+#[napi]
+pub fn transform(opts: JsObject, env: Env) -> napi::Result<JsUnknown> {
+  let config: parcel_js_swc_core::Config = env.from_js_value(opts)?;
+
+  let result = parcel_js_swc_core::transform(config, None)?;
+  env.to_js_value(&result)
+}
+
+#[napi]
+pub fn transform_async(opts: JsObject, env: Env) -> napi::Result<JsObject> {
+  let call_macro = if opts.has_named_property("callMacro")? {
+    let func = opts.get_named_property::<JsUnknown>("callMacro")?;
+    if let Ok(func) = func.try_into() {
+      Some(create_macro_callback(func, env)?)
```

Comment:
I'm not sure if this function is sound ; can we talk about what it's doing & try to do a safe implementation?